File: samples/server/petstore/java-vertx-web/src/main/java/org/openapitools/vertxweb/server/api/UserApiImpl.java
package org.openapitools.vertxweb.server.api;

import java.time.OffsetDateTime;
import org.openapitools.vertxweb.server.model.User;

import org.openapitools.vertxweb.server.ApiResponse;

import io.vertx.core.Future;
import io.vertx.core.json.JsonObject;
import io.vertx.ext.web.handler.HttpException;

import java.util.List;
import java.util.Map;

// Implement this class

public class UserApiImpl implements UserApi {
    public Future<ApiResponse<Void>> createUser(User user) {
        return Future.failedFuture(new HttpException(501));
    }

    public Future<ApiResponse<Void>> createUsersWithArrayInput(List<User> user) {
        return Future.failedFuture(new HttpException(501));
    }

    public Future<ApiResponse<Void>> createUsersWithListInput(List<User> user) {
        return Future.failedFuture(new HttpException(501));
    }

    public Future<ApiResponse<Void>> deleteUser(String username) {
        return Future.failedFuture(new HttpException(501));
    }

    public Future<ApiResponse<User>> getUserByName(String username) {
        return Future.failedFuture(new HttpException(501));
    }

    public Future<ApiResponse<String>> loginUser(String username, String password) {
        return Future.failedFuture(new HttpException(501));
    }

    public Future<ApiResponse<Void>> logoutUser() {
        return Future.failedFuture(new HttpException(501));
    }

    public Future<ApiResponse<Void>> updateUser(String username, User user) {
        return Future.failedFuture(new HttpException(501));
    }

}

File: samples/server/petstore/kotlin-misk/src/main/kotlin/org/openapitools/server/api/api/UserApiImpl.kt
package org.openapitools.server.api.api

import jakarta.inject.Inject
import jakarta.inject.Singleton
import jakarta.validation.Valid
import jakarta.validation.constraints.DecimalMax
import jakarta.validation.constraints.DecimalMin
import jakarta.validation.constraints.Email
import jakarta.validation.constraints.Max
import jakarta.validation.constraints.Min
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Pattern
import jakarta.validation.constraints.Size
import misk.web.HttpCall
import misk.web.PathParam
import misk.web.QueryParam
import misk.web.RequestBody
import misk.web.RequestHeader
import org.openapitools.server.api.model.User

/**
 * @TODO("Fill out implementation")
 */
@Singleton
class UserApiImpl @Inject constructor(
): UserApi {

    override fun createUser(@Valid @RequestBody user: User) {
        TODO()
    }

    override fun createUsersWithArrayInput(@Valid @RequestBody user: kotlin.Array<User>) {
        TODO()
    }

    override fun createUsersWithListInput(@Valid @RequestBody user: kotlin.Array<User>) {
        TODO()
    }

    override fun deleteUser(@PathParam("username") username: kotlin.String) {
        TODO()
    }

    override fun getUserByName(@PathParam("username") username: kotlin.String): User {
        TODO()
    }

    override fun loginUser( @QueryParam(value = "username") username: kotlin.String,  @QueryParam(value = "password") password: kotlin.String): kotlin.String {
        TODO()
    }

    override fun logoutUser() {
        TODO()
    }

    override fun updateUser(@PathParam("username") username: kotlin.String, @Valid @RequestBody user: User) {
        TODO()
    }
}
